src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java

```diff
@@ -0,0 +1,149 @@
+
+        assertSampleNamesMatchInputVCF(originalHeader.getSampleNamesInOrder(), sampleNamesOriginalOrdering);
+        final List<String> batchSortedSampleNames = getBatchSortedList(sampleNamesOriginalOrdering, batchSize);
+
+        final VCFHeader remappedHeader = new VCFHeader(newHeaderLines, batchSortedSampleNames);
+        logger.info("Writing the new header with corrected sample names");
+        writer = createVCFWriter(output);
+        writer.writeHeader(remappedHeader);
+        logger.info("Copying the rest of the VCF");
+    }
+
+    private static void assertSampleNamesMatchInputVCF(final ArrayList<String> sampleNamesFromVCF, final List<String> sampleNameFromSampleMapFile) {
+        if(!CollectionUtils.isEqualCollection(sampleNamesFromVCF, sampleNameFromSampleMapFile)){
+            throw new UserException("The sample names in the provided sample name map file do not match the sample names in the provided vcf.\n" +
+                    "It is important this tool is run with exactly the same sample name map file that was used to create the vcf.");
+        }
+    }
+
+    @Override
+    public void apply(final VariantContext variant, final ReadsContext readsContext, final ReferenceContext referenceContext, final FeatureContext featureContext) {
+        writer.add(variant);
+    }
+
+    @Override
+    public void closeTool() {
+        logger.info("Finished writing the new vcf with corrected sampled names.");
+        if( writer != null) {
+            writer.close();
+        }
+    }
+    private void assertThatTheyReallyWantToProceed(){
+        if( !skipCommandLinePrompt ){
+            final Scanner scanner = new Scanner(System.in);
+            System.out.println(
+                    "\n\nYou are about to create a new VCF that has it's header corrected for the sample swapping bug described in https://github.com/broadinstitute/gatk/issues/3682.\n"+ "" +
+                            "You should be certain you want to do this before proceeding.\n" +
+                            "If the following description does not apply to your VCF then the newly generated vcf will be \n\n \t\tHORRIBLY CORRUPTED\n\n" +
+                            "1: your vcf was generated using a GenomicsDBImport released before gatk version 4.beta.6\n" +
+                            "2: you set --batchSize != 0 when running GenomicsDBImport\n" +
+                            "3: your callset was imported in multiple batches, i.e. your number of samples > batchSize\n" +
+                            "4: you supplied the exact same sampleNameMap file and batch size you used in the initial GenomicsDBImport\n\n" +
+
+                            "If you don't know if this applies to you, please contact GATK support at https://gatkforums.broadinstitute.org/gatk/ for assistance\n" +
+                            "Would you like to proceed? You can rerun with --"+SKIP_PROMPT_LONG_NAME+" to skip this check (y/N)");
+            final String next = scanner.next();
```

Comment:
Use `nextLine()` instead of `next()`, so that the program will block until the user actually hits enter.

Reply:
done